lib/AST/ASTMangler.cpp

```diff
@@ -369,6 +369,37 @@ std::string ASTMangler::mangleReabstractionThunkHelper(
   return finalize();
 }
 
+std::string ASTMangler::mangleAutoDiffAssociatedFunctionHelper(
+    StringRef name, AutoDiffAssociatedFunctionKind kind,
+    const SILAutoDiffIndices &indices, bool isLinearMap) {
+  // TODO(TF-20): Make the mangling scheme robust.
+  // TODO(TF-680): Mangle `@differentiable` atttribute requirements as well.
+  beginManglingWithoutPrefix();
+
+  Buffer << "AD__";
+  Buffer << name;
+  Buffer << '_';
```

Comment:
Could these three lines be combined into: `Buffer << "AD__" << name << '_';`

Reply:
Done in https://github.com/apple/swift/pull/26389/commits/c5b2018a91fe40a34c44d4e7a164fc96f19d34b9.